File: backend/src/main/java/com/aiemail/generator/profile/service/ProfileService.java
package com.aiemail.generator.profile.service;

import com.aiemail.generator.auth.entity.UserEntity;
import com.aiemail.generator.profile.dto.*;

public interface ProfileService {
    ProfileResponse getProfile(UserEntity user);
    ProfileResponse updateProfile(UpdateProfileRequest request, UserEntity user);
    void updatePassword(UpdatePasswordRequest request, UserEntity user);
    ProfileResponse updateSettings(UpdateSettingsRequest request, UserEntity user);
}

File: frontend/src/services/profileService.js
import api from './api';

export const profileService = {
  getProfile: async () => {
    const response = await api.get('/api/v1/profile');
    return response.data;
  },

  updateProfile: async (data) => {
    const response = await api.put('/api/v1/profile', data);
    return response.data;
  },

  changePassword: async (data) => {
    const response = await api.put('/api/v1/profile/password', data);
    return response.data;
  },

  updateSettings: async (data) => {
    const response = await api.put('/api/v1/profile/settings', data);
    return response.data;
  }
};

export default profileService;
